lnwallet/btcwallet/btcwallet.go

```diff
@@ -431,42 +431,61 @@ func (b *BtcWallet) PublishTransaction(tx *wire.MsgTx) error {
 	if err := b.wallet.PublishTransaction(tx); err != nil {
 		switch b.chain.(type) {
 		case *chain.RPCClient:
+			// Output was already spent.
 			if strings.Contains(err.Error(), "already spent") {
-				// Output was already spent.
 				return lnwallet.ErrDoubleSpend
 			}
+
+			// Output was already spent.
 			if strings.Contains(err.Error(), "already been spent") {
-				// Output was already spent.
 				return lnwallet.ErrDoubleSpend
 			}
+
+			// Transaction is spending either output that is
+			// missing or already spent.
 			if strings.Contains(err.Error(), "orphan transaction") {
-				// Transaction is spending either output that
-				// is missing or already spent.
+				return lnwallet.ErrDoubleSpend
+			}
+
+			// RBF enabled transaction was rejected for whatever
+			// reason.
+			if strings.Contains(err.Error(), "replacement transaction") {
 				return lnwallet.ErrDoubleSpend
 			}
 
 		case *chain.BitcoindClient:
+			// Output was spent by other transaction already in the
+			// mempool.
 			if strings.Contains(err.Error(), "txn-mempool-conflict") {
-				// Output was spent by other transaction
-				// already in the mempool.
 				return lnwallet.ErrDoubleSpend
 			}
+
+			// RBF enabled transaction did not have enough fee.
 			if strings.Contains(err.Error(), "insufficient fee") {
-				// RBF enabled transaction did not have enough fee.
 				return lnwallet.ErrDoubleSpend
 			}
+
+			// Transaction is spending either output that is
+			// missing or already spent.
 			if strings.Contains(err.Error(), "Missing inputs") {
-				// Transaction is spending either output that
-				// is missing or already spent.
 				return lnwallet.ErrDoubleSpend
 			}
 
 		case *chain.NeutrinoClient:
+			// Output was already spent.
 			if strings.Contains(err.Error(), "already spent") {
-				// Output was already spent.
 				return lnwallet.ErrDoubleSpend
 			}
 
+			// RBF enabled transaction was rejected for whatever
+			// reason.
+			if strings.Contains(err.Error(), "replacement transaction") {
+				return lnwallet.ErrDoubleSpend
```

Comment:
We might want to end up returning a different error in the future for the purpose of detecting an insufficient fee bump, but can likely be deferred.

Reply:
Will this error actually contain `replacement`? Shouldn't we instead check for a reject code here?

Reply:
Hard to check for error code, as both `RejectInvalid`, `RejectInsufficientFee` and `RejectNonstandard` are used for these.

See https://github.com/btcsuite/btcwallet/pull/647/commits/ca06abc72b9cb7ceb1d4531585cf26aeac1560b7